bindings/capacitorjs/android/src/main/java/org/iota/iotawalletmobile/IotaWalletMobile.java

```diff
@@ -0,0 +1,187 @@
+import org.iota.types.CoinType;
+import org.iota.types.WalletConfig;
+import org.iota.types.events.Event;
+import org.iota.types.events.EventListener;
+import org.iota.types.events.transaction.TransactionProgressEvent;
+import org.iota.types.events.wallet.WalletEventType;
+import org.iota.types.exceptions.WalletException;
+import org.iota.types.secret.StrongholdSecretManager;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import static org.iota.api.NativeApi.callBaseApi;
+import static org.iota.api.NativeApi.callListen;
+
+
+@CapacitorPlugin(name = "IotaWalletMobile")
+public class IotaWalletMobile extends Plugin {
+
+    Wallet wallet = null;
+
+    @PluginMethod()
+    public void messageHandlerNew(final PluginCall call) throws JSONException {
+
+        if (!call.getData().has("clientOptions")
+                || !call.getData().has("storagePath")
+                || !call.getData().has("coinType")
+                || !call.getData().has("secretManager")) {
+            call.reject("clientOptions, storagePath, coinType and secretManager are required");
+        }
+        JSObject _clientOptions = call.getObject("clientOptions");
+        String _storagePath = call.getString("storagePath");
+        Integer _coinType = call.getInt("coinType");
+        JSObject _secretManager = call.getObject("secretManager");
+        String path = getContext().getFilesDir() + _storagePath;
+        Log.d("_clientOptions", _clientOptions.get("nodes").toString());
+
+
+        JSONArray _nodes = _clientOptions.getJSONArray("nodes");
+        String[] nodes = new String[_nodes.length()];
+        for (int i = 0; i < _nodes.length(); i++) {
+            JSONObject _node = (JSONObject) _nodes.get(i);
+            nodes[i] = _node.get("url").toString();
+        }
+        
+        JSONObject secretManager = _secretManager.getJSONObject("stronghold");
+        Log.d("_storagePath", path + secretManager.get("snapshotPath"));
+        if (_coinType == null) {
+            return;
+        }
+        CoinType coinType = CoinType.Shimmer;
+        if (CoinType.Iota.getCoinTypeValue() == _coinType) {
+            coinType = CoinType.Iota;
+        }
+
+        try {
+            wallet = new Wallet(new WalletConfig()
+                    .withClientOptions(new ClientConfig().withNodes(nodes))
+                    .withStoragePath(path)
+                    .withSecretManager(
+                            new StrongholdSecretManager(
+                                    null,
+                                    null,
+                                    path + "/wallet.stronghold"
+                            )
+                    )
+                    .withCoinType(coinType)
+            );
+            JSObject ret = new JSObject();
+            ret.put("messageHandler", 1);
```

Comment:
1 signals the id of the messageHandler returned by the rust side. This is irrelevant for the Java side, but required on the Swift side (since it works with pointers?). Let's add a named variable to signal this in the code.

Reply:
Yes it return a pointer and also it's relevant to the JS side of Firefly.  Added the comment and named var